app/features/TransactionSlice.ts

```diff
@@ -82,8 +77,111 @@ let latestLoadingRequestId: string | undefined;
 const forceLock = new Mutex();
 
 /**
- * Load transactions from storage.
- * Filters out reward transactions based on the account's transaction filter.
+ * Get any transactions that are newer than the newest transaction in the local state. The
+ * transactions are retrieved from the wallet proxy in ascending order, providing it with the
+ * current maximum id of a transaction in the local state. The result is reversed to keep the
+ * sorting (on id) consistently descending.
+ * @param account the account to get new transactions for
+ * @param transactions the current transactions in the state
+ * @param limit the maximum number of transactions to ask the wallet proxy for
+ */
+async function getNewTransactions(
+    account: Account,
+    transactionsInState: TransferTransaction[],
+    limit: number
+): Promise<TransferTransaction[]> {
+    const maxId = transactionsInState
+        .map((t) => t.id)
+        .filter(isDefined)
+        .reduce<string | undefined>(
+            (max, cur) => (!max || max < cur ? cur : max),
+            undefined
+        );
+
+    const transactions: IncomingTransaction[] = [];
+    let full = true;
+    let currentId = maxId;
+
+    // We have to ask for transactions until there are no more. This is needed as we could
+    // receive more than ${limit} transactions in one query to the wallet proxy.
+    while (full) {
+        const transactionsResponseFromWalletProxy = await getTransactionsAscending(
+            account.address,
+            account.transactionFilter,
+            limit,
+            currentId
+        );
+
+        transactions.push(...transactionsResponseFromWalletProxy.transactions);
+        if (transactionsResponseFromWalletProxy.transactions.length === 0) {
+            full = false;
+        } else {
+            currentId =
+                transactionsResponseFromWalletProxy.transactions[
+                    transactionsResponseFromWalletProxy.transactions.length - 1
+                ].id;
+            full = transactionsResponseFromWalletProxy.full;
+        }
+    }
+
+    const transactionsInLocalFormat = transactions.map((txn) =>
+        convertIncomingTransaction(txn, account.address)
+    );
+    const transactionsInDescendingOrder = transactionsInLocalFormat.reverse();
+    return transactionsInDescendingOrder;
+}
+
+/**
+ * Enriches any encrypted transfers (with or without memo) with decrypted amounts, if they
+ * have previously been decrypted and had their amounts stored locally in the database.
+ * @param transactions an array of transactions from the wallet proxy
+ * @returns the transactions enriched with decrypted amounts, and a boolean indicating whether we had decrypted amounts for all the encrypted transfers
+ */
+async function enrichWithDecryptedAmounts(
```

Comment:
It is a little confusing that this and a function in `preload/AccountReport.ts`

Reply:
A little, but as they kind of do the same we will keep it as is.